bigchaindb_driver/connection.py

```diff
@@ -8,29 +9,72 @@
 HttpResponse = namedtuple('HttpResponse', ('status_code', 'headers', 'data'))
 
 
-class Connection:
-    """A Connection object to make HTTP requests."""
+class AbstractConnection(ABC):
+    """Abstract interface for Connection classes
 
-    def __init__(self, *, node_url):
-        """Initializes a :class:`~bigchaindb_driver.connection.Connection`
-        instance.
+    Attributes:
+        node_url (str): URL of this connection
+        session (:class:`~requests.Session`): Long-lived connection to
+            :attr:`node_url`
+    """
+
+    @abstractmethod
+    def __init__(self, node_url):
+        """Initialize a Connnection class to the given node
 
         Args:
-            node_url (str):  Url of the node to connect to.
+            node_url (str): URL of the node to connect to
+        """
+
+    @abstractmethod
+    def request(self, method, path, *, json):
+        """Forward a request to the connected node
+
+        Args:
+            method (str): HTTP method name (e.g.: ``'GET'``)
+            path (str, optional): Path to be appended to the base url of a
+                node for the request.
+            json (dict, keyword, optional): Payload to be sent with the request
+
+        Returns:
+            :class:`~bigchaindb_driver.connection.HttpResponse`: a namedtuple
+            of the response's status code, headers, and data::
+
+                (
+                    'status_code': int,
+                    'headers': dict,
+                    'data': dict|str
+                )
 
+        Raises:
+            :class:`~bigchaindb_driver.exceptions.TransportError`: either a
+            TransportError or a subclass of TransportError based on the
+            response.
+        """
+
+
+class Connection(AbstractConnection):
+    """A Connection object to make HTTP requests."""
+
+    def __init__(self, node_url):
+        """Initializes a :class:`~bigchaindb_driver.connection.Connection`
+        instance. Upon the first request, a long-lived session with the given
+        :attr:`node_url` is created.
+
+        See :meth:`~bigchaindb_driver.connection.AbstractConnection.__init__`
+        for the arguments.
         """
         self.node_url = node_url
         self.session = Session()
 
-    def request(self, method, *, path=None, json=None, **kwargs):
-        """Performs an HTTP requests for the specified arguments.
+    def request(self, method, path=None, *, json=None, **kwargs):
```

Comment:
See commit message in https://github.com/bigchaindb/bigchaindb-driver/pull/74/commits/7dc6958557db3d416c9a6470c9d9f08f6aa3b044